addon/components/validated-input/component.js

```diff
@@ -0,0 +1,67 @@
+// This component is derived from ember-cp-validations. See https://github.com/offirgolan/ember-cp-validations for more information
+import Ember from 'ember';
+import layout from './template';
+const {
+    isEmpty,
+    computed,
+    defineProperty,
+} = Ember;
+
+export default Ember.Component.extend({
+    layout,
+    classNames: ['validated-input'],
+    classNameBindings: ['showErrorClass:has-error', 'isValid:has-success'],
+    model: null,
+    value: null,
+    type: 'text',
+    //inputType === 'text' by default. Can be to set to other types if needed.
+    //Currently supports 'text', 'textarea' and 'date'.
+    inputType: 'text',
+    valuePath: '',
+    placeholder: '',
+    validation: null,
+    isTyping: false,
+
+    init() {
+        this._super(...arguments);
+        var valuePath = this.get('valuePath');
+        defineProperty(this, 'validation', computed.oneWay(`model.validations.attrs.${valuePath}`));
+        defineProperty(this, 'value', computed.alias(`model.${valuePath}`));
+    },
+
+    isTextArea: computed('inputType', function () {
+        return (this.get('inputType') === 'textarea');
+    }),
+    isTextField: computed('inputType', function () {
+        return (this.get('inputType') === 'text');
+    }),
+    isDateField: computed('inputType', function () {
+        return (this.get('inputType') === 'date');
+    }),
```

Comment:
These could be made simpler with the [`equal` computed macro](https://emberjs.com/api/ember/2.8/namespaces/Ember.computed/methods/equal?anchor=equal), e.g.
```
isTextArea: computed.equal('inputType', 'textarea'),
```